utils/azureCredentialsManager.ts

```diff
@@ -135,4 +135,292 @@ export class AzureCredentialsManager {
+        const resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+        if (context) {
+            username = node.userName;
+            password = node.password;
+        } else if (registry.adminUserEnabled) {
+            let creds = await client.registries.listCredentials(resourceGroup, registry.name);
+            password = creds.passwords[0].value;
+            username = creds.username;
+        } else {
+            //grab the access token to be used as a password, and a generic username
+            let creds = await this.getTokens(registry);
+            password = creds.accessToken;
+            username = '00000000-0000-0000-0000-000000000000';
+        }
+        return { password, username };
+    }
+
+    /**
+     * function to allow user to pick a desired image for use
+     * @param repository the repository to look in
+     * @returns an AzureImage object (see azureUtils.ts)
+     */
+    public async getImage(repository: Repository): Promise<AzureImage> {
+        const repoImages: azureUtils.AzureImage[] = await AzureCredentialsManager.getInstance().getImages(repository);
+        console.log(repoImages);
+        let imageList: string[] = [];
+        for (let tempImage of repoImages) {
+            imageList.push(tempImage.tag);
+        }
+        let desiredImage = await vscode.window.showQuickPick(imageList, { 'canPickMany': false, 'placeHolder': 'Choose the image you want to delete' });
+        if (desiredImage === undefined) { return; }
+        let image = repoImages.find((myImage): boolean => { return desiredImage === myImage.tag });
+        if (image === undefined) { return; }
+        return image;
+    }
+
+    /**
+     * function to allow user to pick a desired repository for use
+     * @param registry the registry to choose a repository from
+     * @returns a Repository object (see azureUtils.ts)
+     */
+    public async getRepository(registry: Registry): Promise<Repository> {
+        const myRepos: azureUtils.Repository[] = await AzureCredentialsManager.getInstance().getAzureRepositories(registry);
+        let rep: string[] = [];
+        for (let repo of myRepos) {
+            rep.push(repo.name);
+        }
+        let desiredRepo = await vscode.window.showQuickPick(rep, { 'canPickMany': false, 'placeHolder': 'Choose the repository from which your desired image exists' });
+        if (desiredRepo === undefined) { return; }
+        let repository = myRepos.find((currentRepo): boolean => { return desiredRepo === currentRepo.name });
+        if (repository === undefined) {
+            vscode.window.showErrorMessage('Could not find repository. Check that it still exists!');
+            return;
+        }
+        return repository;
+    }
+
+    /**
+     * function to let user choose a registry for use
+     * @returns a Registry object
+     */
+    public async getRegistry(): Promise<Registry> {
```

Comment:
These are input files, I would keep the outside of the azure credentials manager as its objective is to talk directly to the user, not provide azure usability. Another file may be best, something like azurePrompts